### PR TITLE
Implement sticker messages

### PR DESCRIPTION
**Pull request by olyaluts**

### CODE REVIEW

File: QMChatService/QMChatService/Categories/QBChatMessage+QMCustomParameters.h [hunk 1]
```diff
@@ -31,6 +31,8 @@
 @property (strong, nonatomic) NSString *dialogPhoto;
 @property (strong, nonatomic) NSDate *dialogUpdatedAt;
 
+@property (strong, nonatomic) NSString *stickerMessage;
```
Comment (Raikerian) [inline QMChatService/QMChatService/Categories/QBChatMessage+QMCustomParameters.h:34]: Is this sticker ID string? If yes, stickerID as a name would fit it better :)
Also, you need to serialize/deserialize it, see [implementation](https://github.com/QuickBlox/q-municate-services-ios/blob/25beeebd951476b325f4c07e9f103932fd180774/QMChatService/QMChatService/Categories/QBChatMessage%2BQMCustomParameters.m) of already existent properties.

File: QMChatService/QMChatService/Categories/QBChatMessage+QMCustomParameters.m [hunk 2]
```diff
@@ -330,4 +330,9 @@ - (BOOL)isMediaMessage {
 	return self.attachments.count > 0 || self.attachmentStatus == QMMessageAttachmentStatusLoading;
 }
 
+- (BOOL)isStickerMessage {
```
Comment (Raikerian) [inline QMChatService/QMChatService/Categories/QBChatMessage+QMCustomParameters.m:349]: [isNotificationMessage](https://github.com/908Inc/q-municate-services-ios/blob/development/QMChatService/QMChatService/Categories/QBChatMessage%2BQMCustomParameters.m#L323) getter should be updated now. Sticker is not notification.

File: QMChatService/QMChatService/QMChatService.m [hunk 3]
```diff
@@ -1160,6 +1160,15 @@ - (void)sendAttachmentMessage:(QBChatMessage *)attachmentMessage
     }
 }
 
+- (void)sendStickerMessage:(QBChatMessage *)message
+                  toDialog:(QBChatDialog *)dialog
+             saveToHistory:(BOOL)saveToHistory
+             saveToStorage:(BOOL)saveToStorage
+                completion:(QBChatCompletionBlock)completion
+{
+    [self sendMessage:message type:QMMessageTypeSticker toDialog:dialog saveToHistory:saveToHistory saveToStorage:saveToStorage completion:completion];
```
Comment (Raikerian) [inline QMChatService/QMChatService/QMChatService.m:1170]: AFAIK message, that was received by this method), should contain stickerID in it. I suggest you to add assert for it to be nonnull, as some of developers may experiencing unexpected problems if they will try to send sticker message with nil sticker ID.
Example: NSAssert(message.stickerID != nil, @"Sticker ID required!");

**Comment by Raikerian** [comment]
I will post an updated spec requirements in a main pull request (samples repository) in next few hours.

File: QMChatService/QMChatService/Categories/QBChatMessage+QMCustomParameters.h (hunk 1 above)
Comment (olyaluts) [inline QMChatService/QMChatService/Categories/QBChatMessage+QMCustomParameters.h:34]: In our terms this string called "Sticker Message", it's named like that everywhere in StickerPipe. We have Sticker ID also which is included in the Sticker Message. Example:" [[contentId]]"

**Comment by olyaluts** [comment]
@Raikerian  I've updated pull request based on your input

**Pull request closed without merging**